### PR TITLE
Composition Function Events and Status Conditions

### PR DESCRIPTION
**Pull request by dalton-hill-0**
### Description of your changes

Fixes Crossplane [5402]( https://github.com/crossplane/crossplane/issues/5402)
One-pager PR that documents this effort (Crossplane [5426](https://github.com/crossplane/crossplane/pull/5426))
Should be merged with Crossplane [5450](https://github.com/crossplane/crossplane/pull/5450)

Additional changes: Adds some go packages to be included in the Make logic so that they get picked up by `make reviewable`.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- added unit tests
- tested in a composition function

### CODE REVIEW

File: response/response.go
Comment (negz) [inline response/response.go:174]:
```suggestion
func (o *ResultOption) TargetCompositeAndClaim() *ResultOption {
```

This is a bit of a mouthful, but I think it might be best to use the same naming in the protobuf and API.

File: response/response.go
Comment (negz) [inline response/response.go:37]:
```suggestion
// ResultOption allows you to further configure the Result that gets added to
// the response.
type Result struct {
```

Would just `Result` make sense here?

File: response/response.go
Comment (dalton-hill-0) [reply response/response.go]: updated to `TargetCompositeAndClaim`

File: response/response.go
Comment (dalton-hill-0) [reply response/response.go]: updated to `Result` and changed the description of the struct to match Result in pb

File: response/response.go
Comment (negz) [inline response/response.go:180]:
```suggestion
func (o *Result) ConditionTrue(typ, reason string) *Result {
```

I'm usually all for short variables when the type provides a hint as to what they are (e.g. `t xpv1.ConditionType`). Here where they're both `string` I think longer names are warranted.

Unfortunately we can't use `type` since it's a reserved word in Go. We could use `xpv1.ConditionType` (a subtype of `string`) if we wanted. The downside is folks would need to type convert every time they called this method.

**Review by negz — COMMENTED** [review]
Thanks @dalton-hill-0! One last comment, but this LGTM.

Holding off on approving for now until we get the equivalent c/c changes implemented.

**Comment by negz** [comment]
@dalton-hill-0 This one could use a git history cleanup before I merge too.

**Review by negz — APPROVED** [review]

**Comment by dalton-hill-0** [comment]
@negz 

> This one could use a git history cleanup before I merge too.

Done

**Pull request merged**
